api/service/networkinfo/service.go

```diff
@@ -133,7 +133,9 @@ func (s *Service) DoService() {
 				}
 				p := p2p.Peer{IP: ip, Port: port, PeerID: peer.ID, Addrs: peer.Addrs}
 				utils.GetLogInstance().Info("Notify peerChan", "peer", p)
-				s.peerChan <- p
+				if s.peerChan != nil {
+					s.peerChan <- p
+				}
```

Comment:
else {
  utils..... Error("some error message because should not be nil")
}

Reply:
it's possible to be nil if there is no need to do peer discovery. and no need to send data to a nil channel. This is just a protection.